pkg/apis/ceph.rook.io/v1/network_test.go

```diff
@@ -42,6 +42,23 @@ func TestNetworkCephSpecLegacy(t *testing.T) {
 	assert.Equal(t, expected, net)
 }
 
+func TestValidateNetworkSpec(t *testing.T) {
+	net := NetworkSpec{
+		HostNetwork: true,
+		Provider:    NetworkProviderDefault,
+	}
+	err := ValidateNetworkSpec("", net)
+	assert.Nil(t, err)
+	net = NetworkSpec{
+		HostNetwork: true,
+		Provider:    NetworkProviderHost,
+	}
+	err = ValidateNetworkSpec("", net)
+	//assert.err(t, err)
+	assert.True(t, err != nil)
+
+}
```

Comment:
Overall, I always want to champion adding unit tests. It's a huge win for reliability, and I am happy to see that you take it seriously! 

I think smaller PRs like this are a really good opportunity to build up some of the baseline knowledge about what Go test semantics tend to look like, as well as the patterns Rook uses internally.

- try to keep subtests visually separated by empty lines for easier at-a-glance reading
- also try to avoid empty lines at the end of methods
- when testing for errors, use testify's `assert.Error(t, err)` or `assert.NoError(t, ett)` to improve at-a-glance reading
- try to caputure the full matrix of value permutations that can arise. I don't see any tests checking what happens when `.HostNetwork == false`, which is an important part of the overall logic and helps check that we don't break users with valid configurations
 
I think I captured all these points here to illustrate.
```suggestion
func TestValidateNetworkSpec(t *testing.T) {
	net := NetworkSpec{
		HostNetwork: true,
		Provider:    NetworkProviderDefault,
	}
	err := ValidateNetworkSpec("", net)
	assert.NoError(t, err)

	net = NetworkSpec{
		HostNetwork: true,
		Provider:    NetworkProviderHost,
	}
	err = ValidateNetworkSpec("", net)
	assert.Error(t, err)
	
	net := NetworkSpec{
		HostNetwork: false,
		Provider:    NetworkProviderDefault,
	}
	err := ValidateNetworkSpec("", net)
	assert.NoError(t, err)
	
	net = NetworkSpec{
		HostNetwork: false,
		Provider:    NetworkProviderHost,
	}
	err = ValidateNetworkSpec("", net)
	assert.NoError(t, err)
}
```

I would also suggest using `t.Run("subtest name", func(/*etc*/){}) to run each test as a subtest. While that isn't really necessary for such a small set of tests, I think it is something that Go devs should know how to do, and learning how to do it here for small, easy tests will help later when your unit tests get more complex.

In a similar vein, I would also encourage you to look into the [Table Driven Tests Pattern](https://dave.cheney.net/2019/05/07/prefer-table-driven-tests). These work great for simple functions like this one, and you'll see them quite a bit throughout Rook's code. Also, FWIW, VSCode can auto-generate a table driven test skeleton automatically as a starting place, which I find hugely helpful. You might find similar tools for your environment.

For `IsHost()`, that generated skeleton looks like this. I haven't modified it at all. It's a little much for what we need here, and I would clean it up a bit for the actual test so the boilerplate isn't so intimidating to read at a glance.

```go
func TestNetworkSpec_IsHost(t *testing.T) {
	type fields struct {
		Provider            NetworkProviderType
		Selectors           map[CephNetworkType]string
		AddressRanges       *AddressRangesSpec
		Connections         *ConnectionsSpec
		HostNetwork         bool
		IPFamily            IPFamilyType
		DualStack           bool
		MultiClusterService MultiClusterServiceSpec
	}
	tests := []struct {
		name   string
		fields fields
		want   bool
	}{
		// TODO: Add test cases.
	}
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			n := &NetworkSpec{
				Provider:            tt.fields.Provider,
				Selectors:           tt.fields.Selectors,
				AddressRanges:       tt.fields.AddressRanges,
				Connections:         tt.fields.Connections,
				HostNetwork:         tt.fields.HostNetwork,
				IPFamily:            tt.fields.IPFamily,
				DualStack:           tt.fields.DualStack,
				MultiClusterService: tt.fields.MultiClusterService,
			}
			if got := n.IsHost(); got != tt.want {
				t.Errorf("NetworkSpec.IsHost() = %v, want %v", got, tt.want)
			}
		})
	}
}
```